component/newstorage/batchedstore/batchedstore.go

```diff
@@ -0,0 +1,360 @@
+		batchSizeLimit,
+		&sync.RWMutex{},
+	}
+}
+
+// OpenStore opens a store with the given name and returns a handle.
+// If the store has never been opened before, then it is created.
+// Store names are not case-sensitive. If name is blank, then an error will be returned by the underlying provider.
+func (p *Provider) OpenStore(name string) (newstorage.Store, error) {
+	storeName := strings.ToLower(name)
+
+	p.Lock()
+	defer p.Unlock()
+
+	openStore, ok := p.openStores[storeName]
+	if !ok {
+		underlyingStore, err := p.underlyingProvider.OpenStore(storeName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open store in underlying provider: %w", err)
+		}
+
+		newStore := store{
+			storeName,
+			underlyingStore,
+			make([]newstorage.Operation, 0),
+			p.batchSizeLimit,
+			p.removeStore,
+			&sync.RWMutex{},
+		}
+		p.openStores[storeName] = &newStore
+
+		return &newStore, nil
+	}
+
+	return openStore, nil
+}
+
+// SetStoreConfig sets the configuration on a store.
+// The store must be created prior to calling this method.
+// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned by the underlying provider.
+// If name is blank, then an error will be returned by the underlying provider.
+func (p *Provider) SetStoreConfig(name string, config newstorage.StoreConfiguration) error {
+	err := p.underlyingProvider.SetStoreConfig(name, config)
+	if err != nil {
+		return fmt.Errorf("failed to set store config in underlying provider: %w", err)
+	}
+
+	return nil
+}
+
+// GetStoreConfig gets the current store configuration.
+// The store must be created prior to calling this method.
+// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned by the underlying provider.
+// If name is blank, then an error will be returned by the underlying provider.
+func (p *Provider) GetStoreConfig(name string) (newstorage.StoreConfiguration, error) {
+	config, err := p.underlyingProvider.GetStoreConfig(name)
+	if err != nil {
+		return newstorage.StoreConfiguration{},
+			fmt.Errorf("failed to get store config from underlying provider: %w", err)
+	}
+
+	return config, nil
+}
+
+// Close closes all stores created under this store provider.
+// For persistent store implementations, this does not delete any data in the underlying stores.
+func (p *Provider) Close() error {
+	var openStoresSnapshot []*store
+
+	p.Lock()
+	for _, openStore := range p.openStores {
+		openStoresSnapshot = append(openStoresSnapshot, openStore)
+	}
+	p.Unlock()
+
+	for _, openStore := range openStoresSnapshot {
+		err := openStore.Close()
+		if err != nil {
+			return fmt.Errorf(`failed to close open store with name "%s": %w`, openStore.name, err)
+		}
+	}
+
+	err := p.underlyingProvider.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close underlying provider: %w", err)
+	}
+
+	return nil
+}
+
+// Flush forces all open stores to flush their data immediately.
+func (p *Provider) Flush() error {
```

Comment:
the flush should not be part of provider

Reply:
@troyronda @bstasyszyn  @sandrask

Reply:
The reason this is there is for edge-agent in Trustbloc - it needs that ability to force a flush at times (which you discovered during the last milestone). It needs to flush all the stores. If it's not a part of the Provider, I'm not sure how I can do that.

Reply:
is this method part of Provider interface ?

Reply:
@DRK3

Reply:
@fqutishat No.

Reply:
@fqutishat It's part of the interface now.